src/Analyzers/CSharp/Tests/UseInterpolatedString/CSharpUseInterpolatedStringTests.cs

```diff
@@ -0,0 +1,245 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.UseInterpolatedString;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UseInterpolatedString;
+
+using VerifyCS = CSharpCodeFixVerifier<
+    CSharpUseInterpolatedStringDiagnosticAnalyzer,
+    CSharpUseInterpolatedStringCodeFixProvider>;
+
+public sealed class CSharpUseInterpolatedStringTests
+{
+    [Fact]
+    public Task KeepInterpolatedStrings() => new VerifyCS.Test
+    {
+        TestCode = """
+        class Program
+        {
+            private string MyString => $"My{42}String";
+        }
+        """,
+    }.RunAsync();
+
+    [Fact]
+    public Task KeepInterpolatedVerbatimStrings() => new VerifyCS.Test
+    {
+        TestCode = """
+        class Program
+        {
+            private string MyString1 => @$"My{42}String";
+            private string MyString2 => $@"My{42}String";
+        }
+        """,
+    }.RunAsync();
+
+    [Fact]
+    public Task KeepInterpolatedSingleLineRawStrings() => new VerifyCS.Test
+    {
+        TestCode = """"
+        class Program
+        {
+            private string MyString1 => $"""My{42}String""";
+            private string MyString2 => $"""My{42}String""";
+        }
+        """",
+        LanguageVersion = LanguageVersion.CSharp11,
+    }.RunAsync();
+
+    [Fact]
+    public Task KeepInterpolatedMultiLineRawStrings() => new VerifyCS.Test
+    {
+        TestCode = """"
+        class Program
+        {
+            private string MyString1 => $"""
+                My{42}String
+                """;
+            private string MyString2 => $"""
+                My{42}String
+                """;
+        }
+        """",
+        LanguageVersion = LanguageVersion.CSharp11,
+    }.RunAsync();
+
+    [Fact]
+    public Task FixRegularStrings() => new VerifyCS.Test
+    {
+        TestCode = """
+        class Program
+        {
+            private string MyString1 => "MyString";
+            private string MyString2 => "My{String";
+            private string MyString3 => "My}String";
+            private string MyString4 => "My{S}tring";
+            private string MyString5 => "My}S{tring";
+        }
+        """,
+        FixedCode = """
+        class Program
+        {
+            private string MyString1 => $"MyString";
+            private string MyString2 => $"My{{String";
+            private string MyString3 => $"My}}String";
+            private string MyString4 => $"My{{S}}tring";
+            private string MyString5 => $"My}}S{{tring";
+        }
+        """,
+        ExpectedDiagnostics =
+        {
+            // /0/Test0.cs(3,33): hidden IDE0330: String can be converted to interpolated string
+            VerifyCS.Diagnostic().WithSpan(3, 33, 3, 43),
```

Comment:
Please use test markup for diagnostic locations. Just surround the location with `[|` and `|]` and remove these manually specified `ExpectedDiagnostics`, this should do the trick